nixos/modules/services/system/nix-daemon.nix

```diff
@@ -198,6 +198,7 @@ in
             IOSchedulingClass = cfg.daemonIOSchedClass;
             IOSchedulingPriority = cfg.daemonIOSchedPriority;
             LimitNOFILE = 1048576;
+            Delegate = "yes";
```

Comment:
Should it be conditional on `cfg.settings.use-cgroups`? or does this just not affecting anything if `use-cgroups = false`

Reply:
Making it conditional on `use-cgroups` would be acceptable, although as far as I can tell from the [documentation](https://www.freedesktop.org/software/systemd/man/latest/systemd.resource-control.html#Delegate=), `Delegate` is specific to cgroups and shouldn't affect anything else.

Reply:
That feature can also be automatically enabled, making the detection logic a bit tricky at Nix eval time:
https://github.com/NixOS/nix/blob/46339db18db562607977523aa8cb552a1e9b08d3/src/libstore/globals.hh#L497-L498